internal/services/monitor/monitor_aad_diagnostic_setting_resource.go

```diff
@@ -194,6 +250,89 @@ func resourceMonitorAADDiagnosticSettingCreateUpdate(d *pluginsdk.ResourceData,
 	return resourceMonitorAADDiagnosticSettingRead(d, meta)
 }
 
+func resourceMonitorAADDiagnosticSettingUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Monitor.AADDiagnosticSettingsClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+	log.Printf("[INFO] preparing arguments for Azure ARM AAD Diagnostic Setting.")
+
+	id, err := parse.MonitorAADDiagnosticSettingID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	existing, err := client.Get(ctx, id.Name)
+	if err != nil {
+		return fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+
+	var logs []aad.LogSettings
+	logsChanged := false
+	valid := false
+
+	if !features.FourPointOhBeta() {
+		if d.HasChange("log") {
+			logsChanged = true
+			logs = expandMonitorAADDiagnosticsSettingsLogs(d.Get("log").(*pluginsdk.Set).List())
+			for _, v := range logs {
+				if v.Enabled != nil && *v.Enabled {
+					valid = true
+					break
+				}
+			}
+		}
+	}
+
+	if d.HasChange("enabled_log") {
+		logsChanged = true
+		logs = append(logs, expandMonitorAADDiagnosticsSettingsEnabledLogs(d.Get("enabled_log").(*pluginsdk.Set).List())...)
+		valid = true
+	}
+
+	if !logsChanged && existing.Logs != nil {
+		logs = *existing.Logs
+		for _, v := range logs {
+			if v.Enabled != nil && *v.Enabled {
+				valid = true
+				break
+			}
+		}
+	}
+
+	if !valid {
+		return fmt.Errorf("at least one of the `log` of the %s should be enabled", id)
+	}
```

Comment:
is this check necessary here if we are just taking the logs retrieved from the GET response? Can we assume these will be valid?

Reply:
I think it should be necessary before 4.0 version, we should warn users not to update to all `disabled` log.

Reply:
in this case it looks like we are doing this check when the user has not made changes to the logs

Reply:
Line 296 is checking when logs is not changed, and Line 279 and 289 are checking with logs changed.